src/network/connection.cpp

```diff
@@ -2089,12 +1981,57 @@ void * ConnectionReceiveThread::Thread()
 	return NULL;
 }
 
+bool ConnectionReceiveThread::wait(int timeout_us, fd_set &readset) {
+	// Initialize the set
+	FD_ZERO(&readset);
+	int max_fd = 0;
+	for (std::vector<UDPSocket *>::const_iterator
```

Comment:
Well, I guess it's less likely that you'll accidentally decrease performance more than necessary if you change the socket container type to a list this way.  It's probably slightly less performant with a vector though.  Either way's fine with me.